### PR TITLE
DAOS-7314 Test: Adding EC test case for single data type.

### PR DESCRIPTION
**Pull request by ravalsam**
- Adding EC Rebuild disabled for single data type.

Quick-build: true
Test-tag: ec_disabled_rebuild_single

Signed-off-by: Samir Raval <samir.raval@intel.com>

### CODE REVIEW

**Comment by ravalsam** [comment]
Test results in CI be found from link https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5679/20/artifact/Functional%20Hardware%20Large/erasurecode/

File: src/tests/ftest/util/ec_utils.py [hunk 1]
```diff
@@ -133,3 +142,167 @@ def ior_read_dataset(self, parity=1):
...
+        self.server_count = None
+        self.set_online_rebuild = False
+        self.rank_to_kill = None
+        self.container = []
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super().setUp()
+        self.server_count = len(self.hostlist_servers) * 2
```
Comment (saurabhtandan) [inline src/tests/ftest/util/ec_utils.py:164]: `2` should be replaced by `engine_count = self.server_managers[0].get_config_value("engines_per_host")`

File: src/tests/ftest/util/ec_utils.py [hunk 2]
```diff
@@ -133,3 +142,167 @@ def ior_read_dataset(self, parity=1):
...
+                                   kwargs={"results": self.out_queue})
+        elif operation == 'READ':
+            job = threading.Thread(target=self.read_single_type_dataset,
+                                   kwargs={"results": self.out_queue,
+                                           "parity": parity})
+
+        # Launch the IOR thread
+        job.start()
+
+        # Kill the server rank while IOR in progress
```
Comment (saurabhtandan) [inline src/tests/ftest/util/ec_utils.py:293]: Minor: May be need to remove IOR from comment?

File: src/tests/ftest/erasurecode/ec_offline_rebuild_single.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        """
+        # Write single type data set with all the EC object type
+        self.write_single_type_dataset()
+
+        # Kill the last server rank
+        self.server_managers[0].stop_ranks([self.server_count - 1], self.d_log,
+                                           force=True)
+
+        # Wait for rebuild to start
+        self.pool.wait_for_rebuild(True)
```
[the remark this answers is not in the document]
Comment (ravalsam) [reply src/tests/ftest/erasurecode/ec_offline_rebuild_single.py:43]: good to know, I will remove the lines.

File: src/tests/ftest/erasurecode/ec_offline_rebuild_single.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        # Read data set and verify for different EC object for parity 1 and 2.
+        self.read_single_type_dataset()
+
+        # Kill the another server rank
+        self.server_managers[0].stop_ranks([self.server_count - 2], self.d_log,
+                                           force=True)
+
+        # Wait for rebuild to start
+        self.pool.wait_for_rebuild(True)
```
[the remark this answers is not in the document]
Comment (ravalsam) [reply src/tests/ftest/erasurecode/ec_offline_rebuild_single.py:55]: good to know, I will remove the lines.

File: src/tests/ftest/erasurecode/ec_rebuild_disabled_single.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        self.write_single_type_dataset()
+
+        # Read data set with given all the EC object type
+        self.read_single_type_dataset()
+
+        # Kill the last server rank and wait for 20 seconds,
+        # Rebuild is disabled so data will not be rebuild.
+        self.server_managers[0].stop_ranks(
+            [self.server_count - 1], self.d_log, force=True)
+        time.sleep(20)
```
[the remark this answers is not in the document]
Comment (ravalsam) [reply src/tests/ftest/erasurecode/ec_rebuild_disabled_single.py:45]: Added sleep so have some time after killing the server and before reading it. There is no specific reason to use 20 seconds I just need to wait sometime that's it.

File: src/tests/ftest/util/ec_utils.py (hunk 1 above)
Comment (ravalsam) [reply src/tests/ftest/util/ec_utils.py:164]: I will change

File: src/tests/ftest/util/ec_utils.py (hunk 2 above)
Comment (ravalsam) [reply src/tests/ftest/util/ec_utils.py:293]: yes will fix it.

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Comment by ravalsam** [comment]
Functional Hardware test failure is because of the existing defect DAOS-7350 and not depend on the changes made in this PR.

**Review by rpadma2 — APPROVED** [review]
LGTM.

**Pull request merged**